modules/deploy/main.tf

```
@@ -33,6 +33,12 @@ locals {
 
   script = <<EOF
 #!/bin/bash
+
+if [[ "${var.target_version}" == "${var.current_version != "" ? var.current_version : local.current_version}" ]]; then
```

```suggestion
if [[ '${var.target_version}' == '${var.current_version != "" ? var.current_version : local.current_version}' ]]; then
```

Otherwise weirds things happen when the version is $LATEST (we don't turn on `set -u`, so that would silently be replaced with "")